refactor: use a StorageProperties class to load the GCS properties

jnizet opened the pull request:
This is cleaner than using @Value.
Note that even though the properties must be in lower case with dashes
when reading them, then can be written in camelCase in the yaml file
or in the command-line parameters or environment variables.

Also note that, to be cleaner, I changed the name of the properties:

`globe42.googleCloudStorageCredentials` becomes
`globe42.googleCloudStorage.credentials`, and
`globe42.googleCloudStorageCredentialsPath` becomes
`globe42.googleCloudStorage.credentialsPath`.

In order to avoid forgetting to change the environment when deploying
this new version, I already added the new environment variable in the
Clever Cloud console. We'll just have to remove the old variable when
this change is deployed.

acrepet commented:
Have I had to test this PR @jnizet ? If yes, I don't know exactly how, unless just try to launch the app with the good property names `globe42.googleCloudStorage.credentials` and `globe42.googleCloudStorage.credentialsPath` ;-(

I just have a question : I've well understood that even though the properties must be in lower case with dashes
when reading them, then can be written in camelCase, but why didn't you choose the dashes for the environment variable in the Clever Cloud console  (globe42.google-cloud-storage.credentials) ?

jnizet commented:
> unless just try to launch the app with the good property names globe42.googleCloudStorage.credentials and globe42.googleCloudStorage.credentialsPath

Yes, that's it. And test that you can indeed access the files when doing so. But I added the "to review" label also to make sure you were aware of the change and changed the local launch configurations you might have.

> why didn't you choose the dashes for the environment variable in the Clever Cloud console

No particular reason. Maybe because I knew camelCase worked fine, and wasn't sure if dashes in environment variables would cause problems.

acrepet commented:
OK :+1:
